cmake/Modules/FindLibAJANTV2.cmake

```diff
@@ -1,113 +1,176 @@
-	NAMES ajantv2d libajantv2d
-	HINTS
-		ENV AJASDKPath${_lib_suffix}
-		ENV AJASDKPath
-		ENV DepsPath${_lib_suffix}
-		ENV DepsPath
-		${AJASDKPath${_lib_suffix}}
-		${AJASDKPath}
-		${DepsPath${_lib_suffix}}
-		${DepsPath}
-		${_AJA_NTV2_LIBRARY_DIRS}
-	PATHS
-		/usr/lib /usr/local/lib /opt/local/lib /sw/lib
-	PATH_SUFFIXES
-		lib${_lib_suffix} lib
-		libs${_lib_suffix} libs
-		bin${_lib_suffix} bin
-		../lib${_lib_suffix} ../lib
-		../libs${_lib_suffix} ../libs
-		../bin${_lib_suffix} ../bin)
+find_library(
+  AJA_NTV2_DEBUG_LIB
+  NAMES ajantv2d libajantv2d
+  HINTS ENV
+        AJASDKPath${_lib_suffix}
+        ENV
+        AJASDKPath
+        ENV
+        DepsPath${_lib_suffix}
+        ENV
+        DepsPath
+        ${AJASDKPath${_lib_suffix}}
+        ${AJASDKPath}
+        ${DepsPath${_lib_suffix}}
+        ${DepsPath}
+        ${_AJA_NTV2_LIBRARY_DIRS}
+  PATHS /usr/lib /usr/local/lib /opt/local/lib /sw/lib
+  PATH_SUFFIXES
+    lib${_lib_suffix}
+    lib
+    libs${_lib_suffix}
+    libs
+    bin${_lib_suffix}
+    bin
+    ../lib${_lib_suffix}
+    ../lib
+    ../libs${_lib_suffix}
+    ../libs
+    ../bin${_lib_suffix}
+    ../bin)
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(LibAJANTV2 DEFAULT_MSG AJA_LIBRARIES_INCLUDE_DIR AJA_NTV2_LIB)
+find_package_handle_standard_args(LibAJANTV2 DEFAULT_MSG
+                                  AJA_LIBRARIES_INCLUDE_DIR AJA_NTV2_LIB)
 mark_as_advanced(AJA_LIBRARIES_INCLUDE_DIR AJA_NTV2_LIB)
 
 if(LIBAJANTV2_FOUND)
-	set(AJA_LIBRARIES_INCLUDE_DIR
-		${AJA_LIBRARIES_INCLUDE_DIR}/ajalibraries)
-	set(AJA_LIBRARIES_INCLUDE_DIRS
-		${AJA_LIBRARIES_INCLUDE_DIR}
-		${AJA_LIBRARIES_INCLUDE_DIR}/ajaanc
-		${AJA_LIBRARIES_INCLUDE_DIR}/ajabase
-		${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2
-		${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/includes
-		${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src)
-	if (WIN32)
-		set(AJA_LIBRARIES_INCLUDE_DIRS
-			${AJA_LIBRARIES_INCLUDE_DIRS}
-			${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src/win)
-	elseif (APPLE)
-		set(AJA_LIBRARIES_INCLUDE_DIRS
-			${AJA_LIBRARIES_INCLUDE_DIRS}
-			${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src/mac)
-	elseif(UNIX AND NOT APPLE)
-		set(AJA_LIBRARIES_INCLUDE_DIRS
-			${AJA_LIBRARIES_INCLUDE_DIRS}
-			${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src/lin)
-	endif()
+  set(AJA_LIBRARIES_INCLUDE_DIR ${AJA_LIBRARIES_INCLUDE_DIR}/ajalibraries)
+  set(AJA_LIBRARIES_INCLUDE_DIRS
+      ${AJA_LIBRARIES_INCLUDE_DIR} ${AJA_LIBRARIES_INCLUDE_DIR}/ajaanc
+      ${AJA_LIBRARIES_INCLUDE_DIR}/ajabase ${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2
+      ${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/includes)
+  if(OS_WINDOWS)
+    set(AJA_LIBRARIES_INCLUDE_DIRS
+        ${AJA_LIBRARIES_INCLUDE_DIRS} ${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src
+        ${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src/win)
+  elseif(OS_LINUX)
+    set(AJA_LIBRARIES_INCLUDE_DIRS
+        ${AJA_LIBRARIES_INCLUDE_DIRS} ${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src
+        ${AJA_LIBRARIES_INCLUDE_DIR}/ajantv2/src/lin)
```

Comment:
On Linux like on macOS, we can build ntv2 from source, so those directories does not exist and CMake will complain and fail.

Reply:
Thanks, I suspected as much but didn't have any insight into their include directory structure on Linux.

Reply:
Here an insight from my Arch Linux package:
Build with those command:
``` sh
# Build from source step
mkdir -p build; cd build

cmake \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DAJA_BUILD_OPENSOURCE=ON ..

make

# Install step
cd build

make install DESTDIR="$pkgdir"
```
A sum up of what is installed:
```
/usr/include/ajalibraries/ajaanc/includes/* (many headers)
/usr/include/ajalibraries/ajabase/* (many headers)
/usr/include/ajalibraries/ajantv2/* (many headers)
/usr/lib/libajantv2.a
```